test/core/client_channel/http_proxy_mapper_test.cc

```diff
@@ -157,12 +155,54 @@ TEST(NoProxyTest, WhitespacedEntries) {
 
 // Test entries with invalid CIDR blocks in 'no_proxy' list.
 TEST(NoProxyTest, InvalidCIDREntries) {
-  ScopedSetEnv no_proxy("foo.com, 192.168.0.255/33");
+  ScopedEnvVar no_proxy(NO_PROXY_VAR_NAME, "foo.com, 192.168.0.255/33");
   auto args = ChannelArgs().Set(GRPC_ARG_HTTP_PROXY, "http://proxy.google.com");
   EXPECT_EQ(HttpProxyMapper().MapName("dns:///192.168.1.0:443", &args),
             "proxy.google.com");
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), "192.168.1.0:443");
 }
+
+TEST(ProxyForAddressTest, SetByEnvVar) {
+  ScopedEnvVar address_proxy(HttpProxyMapper::ADDRESS_PROXY_ENV_VAR,
+                             "192.168.0.100:2020");
+  auto args = ChannelArgs();
+  auto address = StringToSockaddr("192.168.0.1:3333");
+  ASSERT_EQ(address.status().code(), absl::StatusCode::kOk);
+  auto mapped = HttpProxyMapper().MapAddress(*address, &args);
+  ASSERT_TRUE(mapped.has_value());
+  EXPECT_EQ(grpc_sockaddr_to_string(&mapped.value(), true),
+            absl::StatusOr<std::string>("192.168.0.100:2020"));
+  EXPECT_EQ(
+      std::string(
```

Comment:
I don't think it's necessary to create a `std::string` here, nor is it necessary to use `value_or()`.  I think this can just be:

```
EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), "192.168.0.1:3333");
```

Same thing in all tests.

Reply:
🤨 I'm most certain there was a reason I added the cast...